dataset/src/main/scala/frameless/IsValueClass.scala

```diff
@@ -0,0 +1,17 @@
+package frameless
+
+import shapeless._
+import shapeless.labelled.FieldType
+
+/** Evidence that `T` is a Value class */
+@annotation.implicitNotFound(msg = "${T} is not a Value class")
+final class IsValueClass[T] private() {}
+
+object IsValueClass {
+  /** Provides an evidence `A` is a Value class */
+  implicit def apply[A <: AnyVal, G <: ::[_, HNil], H <: ::[_ <: FieldType[_ <: Symbol, _], HNil]](
```

Comment:
IsValueClass is required to disambiguate between Value Classes and Primitives. Being a subtype of AnyVal is not enough i.e. `implicitly[Double <:< AnyVal]` (that's a comment for myself).